lib/algos/list/delimiter.js

```diff
@@ -146,6 +143,10 @@ class Delimiter extends Extension {
      *  @return {number}          - indicates if iteration should continue
      */
     filter(obj) {
+        if (this._reachedMaxKeys()) {
+            return FILTER_END;
```

Comment:
When maxKeys is reached, the filtering `obj` would not be listed? So we could miss the `obj`.

Reply:
It was already the case before. It was not added in the result in addContents (this.Contents.push({ key, value}) not called if _reachedMaxKeys true). On that aspect the "fix" is ISO, it does not modify the result value. 

The main difference is we now return FILTER_END as soon as possible, whatever the next entry is. 

Before, if we look in this file (implementation differs from extension to others), the entry may be skipped at line 156, before to be given to addContent/addCommon (where the max keys reached test was). So if the entry which makes the count reached the max keys is followed by skipped entries, we wait to have a valid one before to reurn FILTER_END.

Reply:
Ah yes it is  👍. I did not check the current algo :(